vector/src/main/java/im/vector/app/features/settings/VectorPreferences.kt

```diff
@@ -1020,20 +1020,32 @@ class VectorPreferences @Inject constructor(
         }
     }
 
-    private fun labsSpacesOnlyOrphansInHome(): Boolean {
-        return defaultPrefs.getBoolean(SETTINGS_LABS_SPACES_HOME_AS_ORPHAN, false)
-    }
-
     fun labsAutoReportUISI(): Boolean {
         return defaultPrefs.getBoolean(SETTINGS_LABS_AUTO_REPORT_UISI, false)
     }
 
     fun prefSpacesShowAllRoomInHome(): Boolean {
-        return defaultPrefs.getBoolean(
-                SETTINGS_PREF_SPACE_SHOW_ALL_ROOM_IN_HOME,
-                // migration of old property
-                !labsSpacesOnlyOrphansInHome()
-        )
+        val defaultValue = false
+        return when {
+            defaultPrefs.contains(SETTINGS_PREF_SPACE_SHOW_ALL_ROOM_IN_HOME) -> {
+                defaultPrefs.getBoolean(SETTINGS_PREF_SPACE_SHOW_ALL_ROOM_IN_HOME, defaultValue)
+            }
+            defaultPrefs.contains(SETTINGS_LABS_SPACES_HOME_AS_ORPHAN) -> migrateOrphansInSpacesToShowAllInHome()
+            else -> defaultValue
+        }
+    }
+
+    private fun migrateOrphansInSpacesToShowAllInHome(): Boolean {
+        val showAllRoomsInHome = !labsSpacesOnlyOrphansInHome()
+        defaultPrefs.edit {
+            putBoolean(SETTINGS_PREF_SPACE_SHOW_ALL_ROOM_IN_HOME, showAllRoomsInHome)
+            remove(SETTINGS_LABS_SPACES_HOME_AS_ORPHAN)
+        }
+        return showAllRoomsInHome
+    }
+
+    private fun labsSpacesOnlyOrphansInHome(): Boolean {
+        return defaultPrefs.getBoolean(SETTINGS_LABS_SPACES_HOME_AS_ORPHAN, false)
```

Comment:
Do we have to live with this or can we remove this flag in the next releases?

Reply:
good question! it looks like the orphan flag was added [14 months](https://github.com/vector-im/element-android/pull/3312) ago then replaced [2 months](https://github.com/vector-im/element-android/pull/3765) later 

as it's been over a year I would lean towards removing the migration entirely! what do you think?

Reply:
I totally agree. They (if there are any users) shouldn't be so surprised.

Reply:
removed! https://github.com/vector-im/element-android/pull/6511/commits/6266b2019b01c725299a5d0c6cd02b5249a686ad

Reply:
LGTM! Feel free to merge.